### ISSUE TITLE
Suspicious log

### ISSUE THREAD

**Issue by bmarty** [vector-im/riotX-android#725]
No crash observed, but during sign in, I observe this log: 

```
    java.lang.IllegalStateException: Realm access from incorrect thread. Realm instance can only be closed on the thread it was created.
        at io.realm.BaseRealm.close(BaseRealm.java:558)
        at io.realm.Realm.close(Realm.java:145)
        at kotlin.io.CloseableKt.closeFinally(Closeable.kt:53)
        at im.vector.matrix.android.internal.session.filter.DefaultFilterRepository.storeFilter(DefaultFilterRepository.kt:31)
        at im.vector.matrix.android.internal.session.filter.DefaultFilterRepository$storeFilter$1.invokeSuspend(Unknown Source:12)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:740)
```

**Comment by bmarty** [comment]
Maybe the root cause for sign in again with the same account

**Comment by bmarty** [comment]
Fixed by #726
